### PR TITLE
Introduce stubbed `exclusive` parameter to `File.create`-overridden method

### PR DESCRIPTION
**Pull request by aam**
This is to soften landing of a breaking change https://github.com/dart-lang/sdk/issues/49647

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM. Needs a test exemption from @Hixie

**Comment by aam** [comment]
> LGTM. Needs a test exemption from @Hixie

I added a simple test that verifies that `exclusive` parameter is available. That required bump of a file package dependency.

File: packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart [hunk 1]
```diff
@@ -48,6 +48,11 @@ void main() {
      expect(ErrorHandlingFileSystem.deleteIfExists(file), true);
   });
 
+  testWithoutContext('create accepts exclusive argument', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    fileSystem.file('file').create(exclusive: true);
```
Comment (dnfield) [inline packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart:53]: nit: this should have some kind of expectation in the test :)

**Review by dnfield — APPROVED** [review]
Still LGTM.

File: packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart]: Done!

**Pull request merged**